pkg/config/remote/service/util.go

```diff
@@ -112,13 +124,13 @@ func openCacheDB(path string, agentVersion string) (*bbolt.DB, error) {
 	})
 	if err != nil {
 		_ = db.Close()
-		return recreate(path, agentVersion)
+		return recreate(path, agentVersion, apiKeyHash)
 	}
 
-	if metadata.Version != agentVersion {
+	if metadata.Version != agentVersion || metadata.APIKeyHash != apiKeyHash {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏can you change the log as well